src/App.tsx

```diff
@@ -15,33 +14,121 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const goods = goodsFromServer;
+
+  const [isVisible, setVisible] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
+  const [isSorted, setIsSorted] = useState(false);
+  const [sortBy, setSortBy] = useState('alphabetically');
+
+  const visibleGoodsList = () => setVisible(!isVisible);
+
+  const reverse = () => setIsReversed(!isReversed);
+
+  const reset = () => {
+    setSortBy('');
+    setIsReversed(false);
+    setIsSorted(false);
+  };
+
+  const sortByLength = () => {
+    setIsSorted(true);
+    setSortBy('length');
+  };
+
+  const sortByAlphabetically = () => {
+    setIsSorted(true);
+    setSortBy('alphabet');
+  };
+
+  const newGoods = [...goods];
+
+  if (isSorted) {
+    newGoods.sort((good1, good2): number => {
```

Comment:
BTW, there's no singular form for “goods” in English, it will be better to use word “product”, for example :)

Reply:
Okey